### ISSUE TITLE
[12.0] Feature Request: "Muliples" in Vendor Pricelists

### ISSUE THREAD

**Issue by krister-ts** [odoo/odoo#28935]
Currently, Vendor Pricelists have a field for the Minimum Order Quantity but no field for Multiples!

**Ex:** _Eggs sold in **multiples of 12** @ $0.27EA and **multiples of 18** @ $0.23EA_

Multiples field:
1. Field in the Vendor Pricelist (default 1)
2. A dropdown on the PO form to select Multiple for the line (you may want specific multiples, especially for products in units of mass/volume)

**One slight hurdle I can see:**
The example above can be somewhat complicated if MTO wants to trigger PO for quantity of 31 ...

You would probably want:
Line 1: **18 + 18** [_12+12 costs more than 18 (which is enough)_] @ $0.23
Total: **36** pcs --> $8.28

vs

Line 1: **18** @ $0.23
Line 2: **12** [_since 18 is too many and 12 is smaller cost_] **+ 12** [_need one more, 12 is smallest cost_] @ $0.27
Total: **42** pcs --> $10.62

This latter approach would be ok since a _reasonably intelligent_ human could figure out that _Line 2_ would be better off as another 18 to _Line1_.

**Comment by pedrobaeza** [comment]
Well, this is something very specific and should come together with additions in other parts.

@fgi-odoo what do you think about this wishlist?

**Comment by fgi-odoo** [comment]
Indeed, too much specific. If prices differ depending on packaging, you can create one product ref per packaging type. Those products can be united in the inventory thanks to phantom boms in manufacturing.

**Comment by gdgellatly** [comment]
@fgi-odoo While the request may seem quite specific, it is a fair point worth looking at, particularly if Odoo is serious about verticalisation. 

Vendor pricing is not only fairly vanilla in Odoo it is extremely difficult to extend due to poor inheritance mechanisms in key functions. In fact if I had to nominate one function for worst piece of code ever in Odoo, it would be _select_seller, essentially unchanged in 6 versions.

Some current weaknesses in vendor pricing / selection.
No selection by location/warehouse (e.g. closest supplier).
No break pricing across even a template - e.g. order 500kg or $5000 or 5000 units of one template/category and receive break pricing based on overall quantity.  The same goes for MOQ's, often they are over an entire category, rather than 1 product.
No variable break - e.g. buy 10 get 1 free.
No freight break - e.g. spend over $x and pay no freight.
No vendor rebate support and certainly nothing close to costing to IAS 2 standards where goods are required to be valued at landed cost less expected rebates.
No pluggable mechanism to query vendor stock, or for that matter make a decision based on vendor availability or lead time.
No pass through discount support (well kind of with Purchase requisitions but not really, but can be done with customisation) where a customers negotiates a special price with an upstream supplier. (Naturally also means no pass through rebate)
And then there is a load of other things that go on that aren't even possible manually without some of the above.

All these arrangements are common in any intermediated vertical, e.g. e-commerce, manufacturing, distribution.  So much so, that I'd say buying at the price within the Odoo "way" is actually the least common in any largish business in those industries, particularly if import isn't involved (although that has its own stuff).

Now I'm not suggesting that they all need implementing, but at least providing mechanisms to do so cleanly.  The thing is, procurement is complicated and the price is rarely the price.

**Comment by pedrobaeza** [comment]
@gdgellatly why don't you propose to master a PR with a split of the method for proper inheritance?

**Comment by gdgellatly** [comment]
@pedrobaeza I did last year, at least a basic split on _select_seller but it got rejected. In any case, actually it is a bit harder than that. Its kind of one of those things I know is deeply constraining, but also it is a design change to really fix it that I don't know the answer for. It is actually a really hard problem to cover even for 1 company. Maybe the Odoo AI needs to visit vendor pricing.

**Comment by pedrobaeza** [comment]
Please ping me to the PR you did to give it a check, and any specific improvement you think of, please share. Better little improvements that big changes. Step by step...

**Comment by gdgellatly** [comment]
https://github.com/odoo/odoo/pull/21359

**Comment by pedrobaeza** [comment]
OK, thanks. Today I'm on holidays, but I'll check it next week.

**Comment by krister-ts** [comment]
My need for multiples would be where the Vendor part is equivalent to other vendor parts that all are the same product variant.

**My Use (a minimal version):**

The product variant: **Resistor (0603, 10K Ohm, 1%)**
Vendor Parts for that variant:
1. **RC0603FR-0710KL** from Vendor A (5000pc reel)
2. **CRCW060310K0FKEB** from Vendor B (10000pc reel)

The parts are considered equivalent. However, the different packaging is an issue... for odoo, that is.

@fgi-odoo I have not used the phantom BoM method before but for the THOUSANDS of variants, would you not need thousands of phantom BoMs?

Even allowing a different uom (with the same reference uom for the variant) would be much more straight forward and not really causing issues

For now, what I will do is enter the multiples into the description fields for our purchaser to make the corrections but, honestly, that **sucks**.
